<!DOCTYPE html>
<!--
	Plain-Academic by Vasilios Mavroudis
	Released under the Simplified BSD License/FreeBSD (2-clause) License.
	https://github.com/mavroudisv/plain-academic
-->
<html lang="en">

<head>
	<title>Markus Heinonen</title>
	<meta charset="utf-8">
	<meta name="viewport" content="width=device-width, initial-scale=1">
	<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.6/css/bootstrap.min.css">
	<script src="https://ajax.googleapis.com/ajax/libs/jquery/1.12.0/jquery.min.js"></script>
	<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.6/js/bootstrap.min.js"></script>
	<link href='https://fonts.googleapis.com/css?family=Oswald:700' rel='stylesheet' type='text/css'>
	<link href='https://fonts.googleapis.com/css?family=Philosopher' rel='stylesheet'>
	<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
</head>

<body style="font-family: Georgia, serif; font-size: 1.9em; margin-bottom: 40px; margin-left: 40px;">
	<!-- Page Content -->
	<div class="container" style="margin-top: 60px">
		<div class="row">
			<!-- Entries Column -->
			<div class="col-md-8">
				<!-- Main Image -->
				<div class="col-md-5">
					<img class="img-responsive" src="photo.jpg"><br>
				</div>

				<div class="col-md-7" style="text-align:justify; ">
					<h2 style="font-family: Philosopher, sans-serif;">Markus Heinonen, PhD</h3>
					<p>Academy Research Fellow<br />
					<a href="http://cs.aalto.fi/en">Department of Computer Science</a><br />
					<a href="http://aalto.fi/en">Aalto University</a><br />
					Helsinki, Finland<br><br />
	
					Office: room B360, <a href="https://usefulaaltomap.fi/#!/select/T">CS-building</a><br />
<!--					Address: Konemiehentie 2, 02150 Espoo, Finland <br />-->
					</p>
				</div>

				<div id="projects" class="col-md-12" style="text-align:justify; margin-top:20px">
					<p>I am working with research groups
					<ul>
						<li><a href="http://research.ics.aalto.fi/csb/">CSB</a>: Computational systems biology, Prof. <a
								href="https://users.ics.aalto.fi/harrila/">Harri Lähdesmäki</a></li>
						<li><a href="http://research.cs.aalto.fi/pml/">PML</a>: Probabilistic Machine learning, Profs.
							<a href="https://people.aalto.fi/samuel.kaski">Samuel Kaski</a>, <a
								href="https://users.aalto.fi/~ave/">Aki Vehtari</a></li>
						<li><a href="https://users.aalto.fi/~asolin/group/">MLRG</a>: Machine Learning Research Group, Prof. <a
								href="https://users.aalto.fi/~asolin/">Arno Solin</a></li>
					</ul>

					I visited Prof. <a href="https://perso.telecom-paristech.fr/fdalche/">Florence
						d'Alch&eacute;-Buc</a> in 2013-14 at TeleCom ParisTech, and obtained
					my PhD in 2008-13 from University of Helsinki supervised by Prof. <a
						href="https://people.aalto.fi/juho.rousu">Juho Rousu</a>.
					</p>

					<h3 style="font-family: Philosopher, sans-serif;">Research interests</h3>
					<ul>
						<li>Bayesian deep learning</li>
						<li>Generative modelling: diffusion, normalizing flows</li>
						<li>Learning dynamics: ODEs, SDEs, PDEs</li>
						<li>Uncertainty, calibration, priors, model selection, Gaussian processes</li>
						<li>Bioinformatics: metabolites, proteins, genomes, drugs</li>
					</ul>
				</div>

				<div id="guidelines" class="col-md-10" style="text-align:justify; margin-top:20px">
					<p><i class="fa fa-star" style="color:brown"></i> <a href="advice.html">Advice</a> to machine learning PhD students
					</p>
				</div>
			</div>

			<!-- Links on the Sidebar -->
			<div class="col-md-4" style="margin-top:1%; font-size: 20px; text-align: left;">
				<br />
				<dd><a class="fa fa-envelope" href="mailto:markus.heinonen@gmail.com"> markus.heinonen@gmail.com</a></dd>
				<dd><a href="http://scholar.google.com/citations?user=hFtfHZoAAAAJ" class="fa fa-graduation-cap"> Google Scholar</a></dd>
				<dd><a href="https://arxiv.org/search/cs?searchtype=author&query=Heinonen,+M" class="fa fa-times"> arXiv</a></dd>
				<dd><a href="https://github.com/markusheinonen" class="fa fa-github"> Github</a></dd>
				<dd><a href="https://www.linkedin.com/in/markus-heinonen-4352342/" class="fa fa-linkedin"> LinkedIn</a></dd>
				<dd><a href="assets/cv.pdf" class="fa fa-book"> CV</a></dd>
			</div>

			<!-- Publications -->
			<div class="col-md-10">
				<h3 id="publications" style="font-family: Philosopher, sans-serif;">Publications</h2>
					<ul>
						<b style="margin-left: -2em;">2025</b>


						<li>
							<a href="https://arxiv.org/abs/2506.09648">Scaling Laws for Uncertainty in Deep Learning</a><br>
							Mattia Rosso, Simone Rossi, Giulio Franzese, <u>Markus Heinonen</u>, Maurizio Filippone<br>
							<i>arxiv</i> (2025) &emsp; 
						</li>

						<li>
							<a href="https://arxiv.org/abs/2505.17517">Spacetime Geometry of Denoising in Diffusion Models</a><br>
							Rafal Karczewski, <u>Markus Heinonen</u>, Alison Pouplin, Soren Hauberg, Vikas Garg<br>
							<i>arxiv</i> (2025) &emsp; 
							<a href="https://github.com/rafalkarczewski/spacetime-geometry" class="fa fa-github"> </a> 
						</li>

						<li>
							<a href="https://arxiv.org/abs/2505.17308">Repulsive Ensembles for Bayesian Inference in Physics-informed Neural Networks</a><br>
							Philipp Pilar, <u>Markus Heinonen</u>, Niklas Wahlström<br>
							<i>arxiv</i> (2025) &emsp; 
						</li>

						<li>
							<a href="https://arxiv.org/abs/2505.21813">Optimizing Data Augmentation through Bayesian Model Selection</a><br>
							Madi Matymov, Ba-Hien Tran, Michael Kampffmeyer, <u>Markus Heinonen</u>, Maurizio Filippone<br>
							<i>arxiv</i> (2025) &emsp; 
						</li>

						<li>
							70. VitroBERT: Modeling DILI by Pretraining BERT on In Vitro Data<br>
							Arslan Masood, Anamya Nagaraja, Katia Belaid, Natalie Mesens, Hugo Ceulemans, Samuel Kaski, Dorota Herman, <u>Markus Heinonen</u><br>
							<i>J Chemoinformatics</i> (2025) (to appear) &emsp; 
						</li>

						<li>
							69. <a href="https://openreview.net/forum?id=fameEAljo3">Gaussian Processes with Bayesian Inference of Covariate Couplings</a><br>
							Mattia Rosso, Juho Ylä-Jääski, Zheyang Shen, <u>Markus Heinonen</u>, Maurizio Filippone<br>
							<i>TMLR</i> (2025) &emsp; 
						</li>

						<li>
							68. <a href="https://iopscience.iop.org/article/10.1088/2632-2153/add8e0">Multi-target property prediction and optimization using latent spaces of generative model</a><br>
							Anirudh Jain, <u>Markus Heinonen</u>, Heikki Käsnänen, Julius Sipilä and Samuel Kaski<br>
							<i>Machine Learning: Science and Technology</i> (2025) &emsp; 
						</li>

						<li>
							67. <a href="https://www.arxiv.org/abs/2506.05231">Progressive Tempering Sampling with Diffusion</a><br>
							Severi Rissanen, RuiKang OuYang, Jiajun He, Wenlin Chen, <u>Markus Heinonen</u>, Arno Solin, José Miguel Hernández-Lobato<br>
							<i>ICML</i> (2025) &emsp; 
						</li>

						<li>
							66. <a href="https://arxiv.org/abs/2502.05807">Devil is in the Details: Density Guidance for Detail-Aware Generation with Flow Models</a><br>
							Rafal Karczewski, <u>Markus Heinonen</u>, Vikas Garg<br>
							<i>ICML</i> (2025) &emsp; 
							<a href="https://rafalkarczewski.github.io/blog/2025/diffusion-density/" class="fa fa-home"> </a> 
						</li>

						<li>
							65. <a href="https://www.science.org/doi/10.1126/scitranslmed.adl6758">Single-cell analysis of aplastic anemia reveals a convergence of NK and NK-like CD8+ T cells with a disease-associated TCR signature</a><br>
							Sofie Lundgren, Jani Huuhtanen, ... , Satu Mustjoki<br/>
							<i>Science Translational Medicine</i> 17 (2025) &emsp;
							<a href="https://github.com/janihuuh/aa_manu" class="fa fa-github"> </a>  
						</li>

						<li>
							64. <a href="https://arxiv.org/abs/2310.11439">From Alexnet to Transformers: Measuring the Non-linearity of Deep Neural Networks with Affine Optimal Transport</a><br>
							Quentin Bouniot, Ievgen Redko, Anton Mallasto, Charlotte Laclau, Karol Arndt, Oliver Struckmeier, <u>Markus Heinonen</u>, Ville Kyrki, Samuel Kaski<br/>
							<i>CVPR</i> (2025) &emsp;
							<a href="https://github.com/qbouniot/AffScoreDeep" class="fa fa-github"> </a>  
						</li>
	
						<li>
							63. <a href="https://arxiv.org/abs/2411.01293">Diffusion Models as Cartoonists! The Curious Case of High Density Regions</a><br>
							Rafal Karczewski, <u>Markus Heinonen</u>, Vikas Garg<br>
							<i>ICLR</i> (2025) &emsp;
							<a href="https://github.com/Aalto-QuML/high-density-diffusion" class="fa fa-github"></a> &nbsp; 
							<a href="https://rafalkarczewski.github.io/blog/2025/diffusion-density/" class="fa fa-home"> </a> &nbsp; 
							<a href="assets/cartoon-poster.pdf" class="fa fa-picture-o"></a> &emsp;
						</li>

						<li>
							62. <a href="https://arxiv.org/abs/2410.11149">Free Hunch: Denoiser Covariance Estimation for Diffusion Models Without Extra Costs</a><br>
							Severi Rissanen, <u>Markus Heinonen</u>, Arno Solin<br>
							<i>ICLR</i> (2025) &emsp;
							<a href="https://github.com/severi-rissanen/free-hunch" class="fa fa-github"> </a> &nbsp;
							<a href="assets/hunch-poster.pdf" class="fa fa-picture-o"></a> &emsp;
						</li>

						<li>
							61. <a href="https://arxiv.org/abs/2405.17656">Equivariant Denoisers Cannot Copy Graphs: Align Your Graph Diffusion Models</a><br>
							Najwa Laabid, Severi Rissanen, <u>Markus Heinonen</u>, Arno Solin, Vikas Garg<br>
							<i>ICLR</i> (2025) &emsp;
							<a href="https://github.com/Aalto-QuML/DiffAlign" class="fa fa-github"> </a> &nbsp;
							<a href="https://aalto-quml.github.io/DiffAlign/" class="fa fa-home"> </a> &nbsp; 
							<a href="assets/retrosynthesis-poster.pdf" class="fa fa-picture-o"></a> &emsp;
						</li>

						<li>
							60. <a href="https://arxiv.org/abs/2402.15864">E(3)-equivariant models cannot learn chirality: Field-based molecular generation</a><br>
							Alexandru Dumitrescu, Dani Korpela, <u>Markus Heinonen</u>, Yogesh Verma, Valerii Iakovlev, Vikas Garg, Harri Lähdesmäki<br>
							<i>ICLR</i> (2025) &emsp;
							<a href="https://github.com/Dumitrescu-Alexandru/FMG" class="fa fa-github"> </a>  &nbsp; 
							<a href="https://iclr.cc/media/PosterPDFs/ICLR%202025/28460.png" class="fa fa-picture-o"></a> &emsp;
						</li>

						<li>
							59. <a href="https://arxiv.org/abs/2408.06050">What Ails Generative Structure-based Drug Design: Too Little or Too Much Expressivity?</a><br>
							Rafał Karczewski, Samuel Kaski, <u>Markus Heinonen</u>, Vikas Garg<br>
							<i>AISTATS</i> (2025) &emsp;
							 <a href="https://github.com/rafalkarczewski/simplesbdd" class="fa fa-github"></a> &nbsp;
							 <a href="assets/ssbdd-poster.pdf" class="fa fa-picture-o"></a> &emsp;
							 <i class="fa fa-star" style="color:brown"></i> <font color="brown">Oral presentation (top 2%)</font> 
						</li>
	
						<li>
							58. <a href="https://arxiv.org/abs/2406.01494">Robust Classification by Coupling Data Mollification with Label Smoothing</a><br>
							<u>Markus Heinonen</u>, Ba-Hien Tran, Michael Kampffmeyer, Maurizio Filippone<br>
							<i>AISTATS</i> (2025) &emsp;
							<a href="https://github.com/markusheinonen/supervised-mollification" class="fa fa-github"> </a> &nbsp;
							<a href="assets/mollification-poster.pdf" class="fa fa-picture-o"></a> &emsp;
						</li>

						<li>
							57. <a href="https://doi.org/10.1016/j.bpj.2024.12.010">Probabilistic analysis of spatial viscoelastic cues in 3D cell culture using magnetic microrheometry</a><br>
							Ossi Arasalo, Arttu Lehtonen, Mari Kielosto, <u>Markus Heinonen</u>, Juho Pokki<br>
							<i>Biophysical J</i> (2025) &emsp;
							<a href="https://github.com/arasalo1/spatial_microrheology" class="fa fa-github"> </a>  
						</li>

						<b style="margin-left: -2em;">2024</b>
					
						<li>
							56. <a href="https://doi.org/10.1186/s13321-024-00924-y">Human-in-the-loop Active Learning for Goal-oriented Molecule Generation</a><br>
							Yasmine Nahal, Janosch Menke, Julien Martinelli, <u>Markus Heinonen</u>, Mikhail Kabeshov, Jon Paul Janet, Eva Nittinger, Ola Engkvist, Samuel Kaski <br>
							<i>J Chemoinformatics</i> (2024) &emsp;
							<a href="https://github.com/yasminenahal/hitl-al-gomg" class="fa fa-github"> </a>  
						</li>

						<li>
							55. <a href="https://arxiv.org/abs/2406.16540">Improving robustness to corruptions with multiplicative weight perturbations</a><br>
							Trung Trinh, <u>Markus Heinonen</u>, Luigi Acerbi, Samuel Kaski <br>
							<i>NeurIPS</i> (2024) &emsp;
							<a href="https://github.com/trungtrinh44/DAMP" class="fa fa-github"> </a> &nbsp;
							<a href="https://trungtrinh44.github.io/DAMP/" class="fa fa-home"> </a> &nbsp;
							<a href="https://trungtrinh44.github.io/DAMP/assets/poster.pdf" class="fa fa-picture-o"></a> &emsp;
							<i class="fa fa-star" style="color:brown"></i> <font color="brown">Spotlight presentation (top 8%)</font>
						</li>
						
						<li>
							54. <a href="https://arxiv.org/abs/2404.10024">ClimODE: Climate Forecasting With Physics-informed Neural ODEs</a><br>
							Yogesh Verma, <u>Markus Heinonen</u>, Vikas Garg<br>
							<i>ICLR</i> (2024) &emsp;
							<a href="https://github.com/Aalto-QuML/ClimODE" class="fa fa-github"></a> &nbsp; 
							<a href="https://yogeshverma1998.github.io/ClimODE/" class="fa fa-home"> </a> &nbsp;
							<a href="https://iclr.cc/media/PosterPDFs/ICLR%202024/17438.png" class="fa fa-picture-o"></a> &emsp; 
							<i class="fa fa-star" style="color:brown"></i> <font color="brown">Oral presentation (top 4%)</font>
						</li>

						<li>
							53. <a href="https://arxiv.org/abs/2306.02775">Input-gradient space particle inference for neural network ensembles</a><br>
							Trung Trinh, <u>Markus Heinonen</u>, Luigi Acerbi, Samuel Kaski <br>
							<i>ICLR</i> (2024) &emsp;
							<a href="https://github.com/AaltoPML/FoRDE" class="fa fa-github"> </a>  &nbsp;
							<a href="https://aaltopml.github.io/FoRDE/" class="fa fa-home"> </a>  &nbsp;
							<a href="https://aaltopml.github.io/FoRDE/assets/poster.pdf" class="fa fa-picture-o"> </a> &emsp;
							<i class="fa fa-star" style="color:brown"></i> <font color="brown">Spotlight presentation (top 16%)</font>
						</li>

						<b style="margin-left: -2em;">2023</b>

						<li>
							52. <a href="https://arxiv.org/abs/2307.04110">Learning Space-Time Continuous Latent Neural PDEs from Partially Observed States</a><br>
							Valerii Iakovlev, <u>Markus Heinonen</u>, Harri Lähdesmäki<br>
							<i>NeurIPS</i> (2023) &emsp;
							<a href="https://github.com/yakovlev31/LatentNeuralPDEs" class="fa fa-github"> </a>  &nbsp;
							<a href="https://neurips.cc/media/PosterPDFs/NeurIPS%202023/72590.png" class="fa fa-picture-o"> </a> 							
						</li>

						<li>
							51. <a href="https://arxiv.org/abs/2303.00800">Continuous-Time Functional Diffusion Processes</a><br/>
							Giulio Franzese, Giulio Corallo, Simone Rossi, <u>Markus Heinonen</u>, Maurizio Filippone, Pietro Michiardi</br>
							<i>NeurIPS</i> (2023) &emsp;
							<a href="https://github.com/giulio98/functional-diffusion-processes" class="fa fa-github"> </a>  &nbsp;
							<a href="https://neurips.cc/media/PosterPDFs/NeurIPS%202023/71456.png" class="fa fa-picture-o"> </a>  
						</li>
						<li>
							50. <a href="https://pubs.acs.org/doi/10.1021/acs.chemrestox.2c00378">Chemistry-based
								modelling on phenotype-based drug-induced liver injury annotation: from
								public to proprietary data</a><br />
							Mohammad Moein, <u>Markus Heinonen</u>, Natalie Mesens, Ronnie Chamanza, Chidozie Amuzie,
							Yvonne Will, Hugo Ceulemans, Samuel Kaski, Dorota Herman</br>
							<i>Chemical Research in Toxicology</i> (2023)
						</li>
						<li>
							49. <a href="https://arxiv.org/abs/2305.07500">Learning representations that are closed-form Monge mapping optimal with application to domain adaptation</a><br />
							Oliver Struckmeier, Ievgen Redko, Anton Mallasto, Karol Arndt, <u>Markus Heinonen</u>, Ville Kyrki</br>
							<i>TMLR</i> (2023) &emsp;
							<a href="https://github.com/Oleffa/LaOT" class="fa fa-github"> </a>
						</li>
						<li>
							48. <a href="https://arxiv.org/abs/2304.04726">Uncertainty-Aware Natural Language Inference with Stochastic Weight Averaging</a><br />
							Aarne Talman, Hande Celikkanat, Sami Virpioja, <u>Markus Heinonen</u>, Jörg Tiedemann</br>
							<i>NoDaLiDa</i> (2023) &emsp;
							<a href="https://github.com/Helsinki-NLP/uncertainty-aware-nli" class="fa fa-github"> </a>
						</li>
						<li>
							47. <a href="https://arxiv.org/abs/2306.01005">AbODE: Ab initio antibody design using conjoined ODEs</a><br />
							Yogesh Verma, <u>Markus Heinonen</u>, Vikas Garg </br>
							<i>ICML</i> (2023) &emsp;
							<a href="https://github.com/yogeshverma1998/AbODE_" class="fa fa-github"> </a> &nbsp;
							<a href="https://yogeshverma1998.github.io/AbODE/" class="fa fa-home"> </a>  &nbsp;
							<a href="https://icml.cc/media/PosterPDFs/ICML%202023/23763.png" class="fa fa-picture-o"> </a>  
						</li>
						<li>
							46. <a href="https://arxiv.org/abs/2303.01925">Learning Energy Conserving Dynamics Efficiently with Hamiltonian Gaussian Processes</a><br />
							Magnus Ross, <u>Markus Heinonen</u> </br>
							<i>TMLR</i> (2023) &emsp;
							<a href="https://github.com/magnusross/hgp" class="fa fa-github"> </a>  &nbsp;
							<a href="https://magnusross.github.io/HamiltonianGPs/" class="fa fa-home"> </a>   
						</li>
						<li>
							45. <a href="https://arxiv.org/abs/2207.01234">Incorporating functional summary information in Bayesian neural networks using a Dirichlet process likelihood approach</a><br />
							Vishnu Raj, Tianyu Cui, <u>Markus Heinonen</u>, Pekka Marttinen</br>
							<i>AISTATS</i> (2023) &emsp;
							<a href="https://github.com/v-i-s-h/summary-likelihood" class="fa fa-github"> </a>  &nbsp;
							<a href="https://virtual.aistats.org/media/PosterPDFs/AISTATS%202023/5703.png" class="fa fa-picture-o"> </a> 
						</li>
						<li>
							44. <a href="https://arxiv.org/abs/2210.03466">Latent Neural ODEs with Sparse Bayesian Multiple Shooting</a><br />
							Valerii Iakovlev, Cagatay Yildiz, <u>Markus Heinonen</u>, Harri Lähdesmäki</br>
							<i>ICLR</i> (2023) &emsp;
							<a href="https://github.com/yakovlev31/msvi" class="fa fa-github"> </a>   <br />
						</li>
						<li>
							43. <a href="https://arxiv.org/abs/2206.13397">Generative Modelling With Inverse Heat Dissipation</a> <br />
							Severi Rissanen, <u>Markus Heinonen</u>, Arno Solin</br>
							<i>ICLR</i> (2023) &emsp; 
							<a href="https://github.com/AaltoML/generative-inverse-heat-dissipation" class="fa fa-github"></a> &nbsp;
							<a href="https://aaltoml.github.io/generative-inverse-heat-dissipation/" class="fa fa-home"> </a> &nbsp;
							<a href="https://www.youtube.com/watch?v=qY1juqIDT10" class="fa fa-youtube-play"> </a> 
						</li>

						<b style="margin-left: -2em;">2022</b>
						<li>
							42. <a href="https://jcheminf.biomedcentral.com/articles/10.1186/s13321-022-00667-8">Human-in-the-loop assisted de novo molecular design</a> <br />
							Iiris Sundin, Alexey Voronov, Haoping Xiao, Kostas Papadopoulos, Esben Bjerrum, Markus
							Heinonen, Atanas Patronov, Samuel Kaski, Ola Engkvist</br>
							<i>J Chemoinformatics</i> (2022) &emsp;
							<a href="https://github.com/MolecularAI/reinvent-hitl" class="fa fa-github"> </a>   <br />
						</li>
						<li>
							41. <a href="https://academic.oup.com/bioinformatics/article/39/1/btac788/6881078">TCRconv: predicting recognition between T cell receptors and epitopes using contextualized motifs</a> <br />
							Emmi Jokinen, Alexandru Dumitrescu, Jani Huuhtanen, Vladimir Gligorijevic, Satu Mustjoki, Richard Bonneau, <u>Markus Heinonen</u>, Harri Lähdesmäki </br>
							<i>Bioinformatics</i> (2022) &emsp;
							<a href="https://github.com/emmijokinen/TCRconv" class="fa fa-github"> </a>   <br />
						</li>
						<li>
							40. <a href="https://arxiv.org/abs/2210.06032">Modular Flows: Differential Molecular Generation</a> <br />
							Yogesh Verma, Samuel Kaski, <u>Markus Heinonen</u>, Vikas Garg</br>
							<i>NeurIPS</i> (2022) &emsp;
							<a href="https://github.com/yogeshverma1998/Modular-Flows-Neurips-2022" class="fa fa-github"> </a>  &nbsp; 
							<a href="https://yogeshverma1998.github.io/ModFlow/" class="fa fa-home"> </a>  &nbsp;
							<a href="https://neurips.cc/media/PosterPDFs/NeurIPS%202022/53126.png" class="fa fa-picture-o"></a>  
						</li>
						<li>
							39. <a href="https://arxiv.org/abs/2206.02435">Tackling covariate shift with node-based Bayesian neural networks</a> <br />
							Trung Trinh, <u>Markus Heinonen</u>, Luigi Acerbi, Samuel Kaski</br>
							<i>ICML</i> (2022) &emsp;
							<a href="https://github.com/AaltoPML/node-BNN-covariate-shift" class="fa fa-github"></a> &nbsp;
							<a href="https://aaltopml.github.io/node-BNN-covariate-shift/" class="fa fa-home"></a> &nbsp;
							<a href="https://icml.cc/media/PosterPDFs/ICML%202022/82debd8a12b498e765a11a8e51159440_qExaNRB.png" class="fa fa-picture-o"></a>  &emsp;
							<i class="fa fa-star" style="color:brown"></i> <font color="brown">Oral presentation (top 10%)</font>
						</li>
						<li>
							38. <a href="https://arxiv.org/abs/2106.10905">Variational multiple shooting for Bayesian ODEs with Gaussian processes</a> <br />
							Pashupati Hegde, Cagatay Yildiz, Harri Lähdesmäki, Samuel Kaski, <u>Markus Heinonen</u></br>
							<i>UAI</i> (2022) &emsp;
							<a href="https://github.com/hegdepashupati/gaussian-process-odes" class="fa fa-github"> </a> 
						</li>
						<li>
							37. <a href="https://arxiv.org/abs/2006.10571">Likelihood-Free Inference with Deep Gaussian Processes</a> <br />
							Alexander Aushev, Henri Pesonen, <u>Markus Heinonen</u>, Jukka Corander, Samuel Kaski</br>
							<i>Computational Statistics and Data Analysis</i> (2022) &emsp;
							<a href="https://github.com/alexaushev/LFIwithDGPs" class="fa fa-github"> </a>
						</li>
						<li>
							36. <a href="https://doi.org/10.1186/s12859-022-04734-7"> Modeling binding specificities of transcription factor pairs with random forests</a> <br/>
							Anni Antikainen, <u>Markus Heinonen</u>, Harri Lähdesmäki</br>
							<i>BMC Bioinformatics</i> (2022) &emsp;
							<a href="https://github.com/AnniAleksandra/ComBind" class="fa fa-github"></a>
						</li>
						<b style="margin-left: -2em;">2021</b>
						<li>
							35. <a href="https://doi.org/10.1111/vox.13223">Prediction and impact of personalized donation intervals</a> <br />
							Jarkko Toivonen, Yrjö Koski, Esa Turkulainen, Femmeke Prinsze, Pietro della Briotta Parolo,
							<u>Markus Heinonen</u>, Mikko Arvas</br>
							<i>Vox Sanguinis</i> (2021) &emsp;
							<a href="https://github.com/FRCBS/Hb_predictor_container" class="fa fa-github"></a>
						</li>
						<li>
							34. <a href="https://arxiv.org/abs/2110.03768">De-randomizing MCMC dynamics with the diffusion Stein operator</a> <br />
							Zheyang Shen, <u>Markus Heinonen</u>, Samuel Kaski</br>
							<i>NeurIPS</i> (2021) &emsp;
							<a href="https://github.com/zheyangshen/generalized_stein_variational_gd" class="fa fa-github"></a>
						</li>

						<li>
							33. <a href="https://arxiv.org/abs/2010.09327">Bayesian Inference for Optimal Transport with Stochastic Cost</a> <br />
							Anton Mallasto, <u>Markus Heinonen</u>, Samuel Kaski</br>
							<i>ACML</i> (2021)<br />
						</li>

						<li>
							32. <a href="https://arxiv.org/abs/2102.04764">Continuous-Time Model-Based Reinforcement Learning</a> <br />
							Çağatay Yıldız, <u>Markus Heinonen</u>, Harri Lähdesmäki</br>
							<i>ICML</i> (2021) &emsp;
							<a href="https://github.com/cagatayyildiz/oderl" class="fa fa-github"> </a>   <br />
						</li>
						<li>
							31. <a href="https://www.biorxiv.org/content/10.1101/542332">Predicting recognition between T cell receptors and epitopes with TCRGP</a> <br />
							Emmi Jokinen, <u>Markus Heinonen</u>, Jani Huuhtanen, Satu Mustjoki, Harri Lähdesmäki</br>
							<i>PLOS Computational Biology</i> (2021) &emsp;
							 <a href="https://github.com/emmijokinen/TCRGP" class="fa fa-github"> </a>  
						</li>

						<li>
							30. <a href="https://arxiv.org/abs/2006.08956">Learning continuous-time PDEs from sparse data with graph neural networks</a> <br />
							Valerii Iakovlev, <u>Markus Heinonen</u>, Harri Lähdesmäki</br>
							<i>ICLR</i> (2021) &emsp;
							<a href="https://github.com/yakovlev31/graphpdes_experiments" class="fa fa-github"> </a>   <br />
						</li>

						<li>
							29. <a href="https://arxiv.org/abs/2003.03080">Sparse Gaussian Processes Revisited: Bayesian Approaches to Inducing-Variable Approximations</a> <br />
							Simone Rossi, <u>Markus Heinonen</u>, Edwin Bonilla, Zheyang Shen, Maurizio Filippone</br>
							<i>AISTATS</i> (2021) &emsp;
							<a href="https://github.com/srossi93/Sparse-Gaussian-Processes-Revisited" class="fa fa-github"> </a>   <br />
						</li>

						<b style="margin-left: -2em;">2020</b>
						<li>
							28. <a href="https://link.springer.com/article/10.1007/s00253-020-10960-x">Substrate specificity of 2-Deoxy-D-ribose 5-phosphate aldolase (DERA) assessed by different protein engineering and machine learning methods</a> <br />
							Sanni Voutilainen, <u>Markus Heinonen</u>, Martina Andberg, Emmi Jokinen, Hannu Maaheimo, Johan Pääkkönen, Nina Hakulinen, Juha Rouvinen, Harri Lähdesmäki, Samuel Kaski, Juho Rousu, Merja Penttilä, Anu Koivula</br>
							<i>Applied Microbiology and Biotechnology</i> (2020)</br>
						</li>

						<li>
							<a href="https://arxiv.org/abs/2011.01226">Sample-efficient reinforcement learning using deep Gaussian processes</a> <br />
							Charles Gadd, <u>Markus Heinonen</u>, Harri Lähdesmäki, Samuel Kaski</br>
							<i>Technical report</i> (2020)<br />
						</li>

						<li>
							<a href="http://arxiv.org/abs/2010.13498">Scalable Bayesian neural networks by layer-wise input augmentation</a> <br />
							Trung Trinh, Samuel Kaski, <u>Markus Heinonen</u></br>
							<i>Technical report</i> (2020) &emsp; 
							<a href="https://github.com/trungtrinh44/ibnn" class="fa fa-github"> </a>  <br/>
						</li>

						<li>
							27. <a href="https://arxiv.org/abs/1905.09917">Learning spectrograms with convolutional spectral kernels</a> <br />
							Zheyang Shen, <u>Markus Heinonen</u>, Samuel Kaski</br>
							<i>AISTATS</i> (2020)</br>
						</li>

						<b style="margin-left: -2em;">2019</b>
						<li>
							26. <a href="http://arxiv.org/abs/1905.10994">ODE<sup>2</sup>VAE: Deep generative second
								order ODEs with Bayesian neural
								networks</a> <br />
							Cagatay Yildiz, <u>Markus Heinonen</u>, Harri Lähdesmäki</br>
							<i>NIPS</i> (2019) &emsp;
							<a href="https://github.com/cagatayyildiz/ODE2VAE" class="fa fa-github"> </a> &nbsp;
							<a href="https://github.com/cagatayyildiz/ODE2VAE/blob/master/ode2vae_poster.pdf" class="fa fa-picture-o"></a> &nbsp;
							<a href="https://www.youtube.com/watch?v=PscfJTyELbQ" class="fa fa-youtube-play"></a>
						</li>

						<li>
							25. <a href="https://arxiv.org/abs/1810.03052">Deep Convolutional Gaussian Processes</a> <br />
							Kenneth Blomqvist, Samuel Kaski, <u>Markus Heinonen</u> </br>
							<i>ECML/PKDD</i> (2019) &emsp;
							<a href="https://github.com/kekeblom/DeepCGP" class="fa fa-github"></a>
						</li>

						<li>
							24. <a href="https://arxiv.org/abs/1804.06673">Bayesian Metabolic Flux Analysis reveals
								intracellular flux couplings</a> <br />
							<u>Markus Heinonen</u>, Maria Osmala, Henrik Mannerström, Janne Wallenius, Samuel Kaski,
							Juho Rousu, Harri Lähdesmäki</br>
							<i>ISMB/ECCB</i> (2019) &emsp;
							   <a href="https://github.com/markusheinonen/bamfa" class="fa fa-github"> </a>  
						</li>

						<li>
							23. <a href="https://arxiv.org/abs/1810.04066"> Deep learning with differential Gaussian process flows</a> <br />
							Pashupati Hegde, <u>Markus Heinonen</u>, Harri Lähdesmäki, Samuel Kaski </br>
							<i>AISTATS</i> (2019) &emsp;
							<a href="https://github.com/hegdepashupati/differential-dgp" class="fa fa-github"></a> &nbsp;
							<a href="https://users.aalto.fi/~heinom10/diffgp-poster.pdf" class="fa fa-picture-o"></a> &emsp; 
								<i class="fa fa-star" style="color:brown"></i> <font color="brown">Notable paper award (top 1%)</font>
						</li>

						<li>
							22. <a href="https://arxiv.org/abs/1810.04416">Harmonizable mixture kernels with variational
								Fourier features</a> <br/>
							Zheyang Shen, <u>Markus Heinonen</u>, Samuel Kaski</br>
							<i>AISTATS</i> (2019)
						</li>
						<b style="margin-left: -2em;">2018</b>
						<li>
							21. <a href="https://arxiv.org/abs/1807.05748">Learning Stochastic Differential Equations
								With Gaussian Processes Without Gradient Matching</a> <br />
							Cagatay Yildiz, <u>Markus Heinonen</u>, Jukka Intosalmi, Henrik Mannerström, Harri
							Lähdesmäki</br>
							<i>MLSP</i> (2018) &emsp;
							<a href="https://github.com/cagatayyildiz/npde" class="fa fa-github"></a>  
						</li>

						<li>
							20. <a href="https://arxiv.org/abs/1803.04303">Learning unknown ODE models with Gaussian
								processes</a> <br />
							<u>Markus Heinonen</u>, Cagatay Yildiz, Henrik Mannerström, Jukka Intosalmi, Harri
							Lähdesmäki</br>
							<i>ICML</i> (2018) &emsp;
							    <a href="https://github.com/cagatayyildiz/npde" class="fa fa-github"> </a> &nbsp;
											<a href="https://github.com/cagatayyildiz/npde/blob/master/npode_poster.pdf" class="fa fa-picture-o"></a>
						</li>

						<li>
							19. <a href="https://arxiv.org/abs/1803.05036">Variational zero-inflated Gaussian processes
								with sparse kernels</a> <br />
							Pashupati Hegde, <u>Markus Heinonen</u>, Samuel Kaski</br>
							<i>UAI</i> (2018) &emsp;
							<a href="https://github.com/hegdepashupati/zero-inflated-gp" class="fa fa-github"> ZIGP</a> &nbsp;
							<a href="https://github.com/hegdepashupati/gprn-svi" class="fa fa-github"> GPRN</a>  
						</li>

						<li>
							<a href="https://arxiv.org/abs/1811.10978">Neural Non-Stationary Spectral Kernel</a> <br />
							Sami Remes, <u>Markus Heinonen</u>, Samuel Kaski</br>
							Technical report (2018)
						</li>

						<li>
							18. <a
								href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0204960">
								 Temporal clustering analysis of endothelial cell gene expression following exposure to a conventional radiotherapy dose fraction using Gaussian process clustering</a> <br/>
							<u>Markus Heinonen</u>, Fabien Milliat, Mohamed Benadjaoud, Agnès François, Valérie Buard, Georges Tarlet, Florence d’Alché-Buc, Olivier Guipaud</br>
							<i>PLOS ONE</i> (2018)
						</li>
						<li>
							17. <a href="https://academic.oup.com/bioinformatics/article/34/13/i509/5045738">Learning
								with multiple pairwise kernels for drug bioactivity prediction</a>
							<br />
							Anna Cichonska, Tapio Pahikkala, Sandor Szedmak, Heli Julkunen, Antti Airola, Markus
							Heinonen, Tero Aittokallio, Juho Rousu</br>
							<i>Bioinformatics</i> (2018) &emsp;
							<a href="https://github.com/aalto-ics-kepaco/pairwiseMKL" class="fa fa-github"></a>
						</li>

						<li>
							16. <a href="https://arxiv.org/abs/1802.02852">mGPfusion: Predicting protein stability
								changes with Gaussian process kernel learning and data fusion</a> <br />
							Emmi Jokinen, <u>Markus Heinonen</u>, Harri Lähdesmäki</br>
							<i>Bioinformatics</i> (2018) &emsp;
							<a href="https://github.com/emmijokinen/mgpfusion" class="fa fa-github"> </a>  
						</li>

						<li>
							15. <a href="https://pubs.acs.org/doi/10.1021/acs.jpcb.7b11367">Flex ddG: Rosetta
								Ensemble-Based Estimation of Changes in Protein-Protein Binding
								Affinity Upon Mutation</a> <br />
							Kyle Barlow, Shane Conchuir, Samuel Thompson, Pooja Suresh, James Lucas, Markus
							Heinonen, Tanja Kortemme</br>
							<i>J Physical Chemistry B</i> (2018)
						</li>

						<b style="margin-left: -2em;">2017</b>

						<li>
							14. <a href="https://arxiv.org/abs/1702.08402">A Mutually-Dependent Hadamard Kernel for
								Modelling Latent Variable Couplings</a>
							<br />
							Sami Remes, <u>Markus Heinonen</u>, Samuel Kaski</br>
							<i>ACML</i> (2017) &emsp;
						 <a href="https://github.com/sremes/wishart-gibbs-kernel" class="fa fa-github"> </a>  
						</li>

						<li>
							13. <a href="https://arxiv.org/abs/1705.08736">Non-Stationary Spectral Kernels</a> <br />
							Sami Remes, <u>Markus Heinonen</u>, Samuel Kaski</br>
							<i>NIPS</i> (2017) &emsp;
					  <a href="https://github.com/sremes/nonstationary-spectral-kernels" class="fa fa-github"> </a>  
						</li>

						<b style="margin-left: -2em;">2016</b>
						<li>
							12. <a href="http://proceedings.mlr.press/v63/Brault39.html">Random Fourier Features for
								operator-valued kernels</a> <br />
							Romain Brault, <u>Markus Heinonen</u>, Florence d'Alche-Buc</br>
							<i>ACML</i> (2016) &emsp;
							<a href="https://github.com/operalib/operalib/tree/ORFF" class="fa fa-github"></a>
						</li>

						<li>
							11. <a href="https://arxiv.org/abs/1508.04319">Non-Stationary Gaussian
								Process Regression with Hamiltonian Monte Carlo</a> <br />
							<u>Markus Heinonen</u>, Henrik Mannerström, Juho Rousu, Samuel Kaski, Harri Lähdesmäki</br>
							<i>AISTATS</i> (2016) &emsp;
			    <a href="https://github.com/markusheinonen/adaptivegp" class="fa fa-github"> </a>  
						</li>

						<li>
							10. <a
								href="https://biotechnologyforbiofuels.biomedcentral.com/articles/10.1186/s13068-016-0547-5">Genome
								wide analysis of protein production load in Trichoderma reesei</a> <br />
							Tiina Pakula, Heli Nygren, Dorothee Barth, <u>Markus Heinonen</u>, Sandra Castillo, Merja
							Penttilä, Mikko Arvas</br>
							<i>Biotechnology for Biofuels</i> (2016)
						</li>

						<b style="margin-left: -2em;">2015</b>

						<li>
							9. <a href="http://bioinformatics.oxfordjournals.org/content/31/5/728.abstract">Detecting
								time periods of differential gene expression using Gaussian processes:
								An application to endothelial cells exposed to radiotherapy dose fraction</a> <br />
							<u>Markus Heinonen</u>, Olivier Guipaud, Fabien Milliat, Valerie Buard, Beatrice Micheau,
							Georges Tarlet, Marc Benderittter, Farida Zehraoui, Florence d'Alche-Buc</br>
							<i>Bioinformatics</i> (2015) &emsp;
			    <a href="http://cran.r-project.org/web/packages/nsgp/index.html" class="fa fa-github"></a>  
						</li>

						<b style="margin-left: -2em;">2014</b>
						<li>
							<a href="http://arxiv.org/abs/1411.5172">Learning nonparametric differential equations with
								operator-valued kernels and gradient matching</a> <br />
							<u>Markus Heinonen</u>, Florence d'Alche-Buc</br>
							<i>Technical report</i> (2014)
						</li>

						<b style="margin-left: -2em;">2013</b>
						<li>
							<a
								href="https://www.researchgate.net/publication/280891120_Time-dependent_gaussian_process_regression_and_significance_analysis_for_sparse_time-series">Time-dependent
								gaussian process regression and significance analysis for sparse
								time-series</a> <br />
							<u>Markus Heinonen</u>, Olivier Guipaud, Fabien Milliat, Valerie Buard, Beatrice Micheau,
							Florence d'Alche-Buc</br>
							<i>Machine Learning in Systems Biology</i> (2013)
						</li>

						<li>
							8. <a href="http://www.mdpi.com/2218-1989/3/2/484">Metabolite Identification trough Machine
								Learning -- Tackling CASMI Challenge using FingerID</a> <br />
							Huibin Shen, Nicola Zamboni, <u>Markus Heinonen</u>, Juho Rousu</br>
							<i>Metabolites</i> (2013)
						</li>

						<b style="margin-left: -2em;">2012</b>
						<li>
							<a href="http://meetingorganizer.copernicus.org/EGU2012/EGU2012-11061.pdf">Full waveform
								forward seismic modeling of geologically complex environment:
								Comparison of simulated and field seismic data </a> <br />
							Suvi Heinonen, <u>Markus Heinonen</u> and Emilia Koivisto</br>
							<i>EGU General Assembly</i> (2012)
						</li>

						<li>
							7. <a href="http://bioinformatics.oxfordjournals.org/content/28/18/2333">Metabolite
								identification and fingerprint prediction via machine learning</a>
							<br />
							<u>Markus Heinonen</u>, Huibin Shen, Nicola Zamboni, Juho Rousu</br>
							<i>Bioinformatics</i> (2012)
						</li>

						<li>
							6. <a href="https://www.researchgate.net/publication/267985665_Efficient_Path_Kernels_for_Reaction_Function_Prediction">Efficient path kernels for reaction function prediction</a> <br />
							<u>Markus Heinonen</u>, Niko Välimäki, Veli Mäkinen, Juho Rousu</br>
							<i>International Conference on Bioinformatics Models,
								Methods and Algorithms</i> (2012) </br>
						</li>

						<b style="margin-left: -2em;">2011</b>
						<li>
							5. <a href="http://online.liebertpub.com/doi/abs/10.1089/cmb.2009.0216">Computing atom
								mappings for biochemical reactions without subgraphs isomorphism</a> <br />
							<u>Markus Heinonen</u>, Sampsa Lappalainen, Taneli Mielikäinen, Juho Rousu</br>
							<i>J Computational Biology</i> (2011) &nbsp;&nbsp;
<!--							    <a href="http://www.cs.helsinki.fi/u/mqheinon/atommaps.tar.gz">KEGG 01/2009
								atommappings</a>   <a
								href="http://www.cs.helsinki.fi/u/mqheinon/mapper2011-src.tar.gz">bin + src</a>  -->
						</li>

						<b style="margin-left: -2em;">2010</b>
						<li>
							4. <a href="http://www.springerlink.com/content/mj870200037883k0/">Structured output
								prediction of anti-cancer drug activity</a> <br />
							Hongyu Su, <u>Markus Heinonen</u>, Juho Rousu</br>
							<i>Pattern Recognition in Bioinformatics</i> (2010) </br>
						</li>

						<li>
							3. <a href="http://www.helsinki.fi/pgm2010/papers/su.pdf">Multilabel Classification of
								Drug-like Molecules via Max-margin Conditional Random Fields</a> <br />
							Hongyu Su, <u>Markus Heinonen</u>, Juho Rousu</br>
							<i>Probabilistic Graphical Models</i> (2010) </br>
						</li>

						<b style="margin-left: -2em;">2008</b>
						<li>
							2. <a href="http://onlinelibrary.wiley.com/doi/10.1002/rcm.3701/full">FiD: a software for
								<em>ab initio</em> structural identification of product ions
								from tandem mass spectrometric data</a> <br />
							<u>Markus Heinonen</u>, Ari Rantanen, Taneli Mielikäinen, Juha Kokkonen, Jari Kiuru, Raimo
							Ketola, Juho Rousu</br>
							<i>Rapid Communications in Mass Spectrometry</i> (2008) </br>
						</li>

						<b style="margin-left: -2em;">2006</b>
						<li>
							1. <a href="https://users.aalto.fi/~heinom10/heinonen06.pdf"><em>Ab initio</em> prediction
								of molecular fragments from tandem mass spectrometry data</a> <br />
							<u>Markus Heinonen</u>, Ari Rantanen, Taneli Mielikäinen, Esa Pitkänen, Juha Kokkonen, Juho
							Rousu</br>
							<i>German Conference on Bioinformatics</i> (2006) </br>
						</li>
					</ul>
			</div>
		</div>
	</div>
	<!-- /.container -->
	<!-- Other people may like it too! -->
	<a style="color:#b5bec9;font-size:0.8em; float:right;" href="https://github.com/mavroudisv/plain-academic">Plain
		Academic</a>
</body>

</html>
